biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceUtils.java

```diff
@@ -474,4 +474,12 @@ public static String toProvideCapability(Capability cap) throws Exception {
 		return result;
 	}
 
+	public static boolean isFragment(Resource resource) {
+		List<Capability> capabilities = resource.getCapabilities(IdentityNamespace.IDENTITY_NAMESPACE);
```

Comment:
OK, Taking a larger look at this class, it seems to me the impl should be 

```java
IdentityCapability identity = getIdentityCapability(resource);
if (identity == null) 
  return false;
return indentity.type() == IdentityCapability.Type.fragment;
```

Then the code here uses the other parts of this class rather than being completely isolated from it.

But then I have no idea if the magic reflection code works correctly :-)